### PR TITLE
fix: clearing watch positions doesn't works sometimes

### PR DESCRIPTION
**Pull request by manish99verma**

### CODE REVIEW

**Comment by Bnyro** [comment]
This PR assumes that the timer task is still running in the background even though the player fragment already got killed?

I think it'd be easier to just release the timer in the `onDestroy` method to be sure and I don't see the need to create a new `CustomTimer` class to be honest, it doesn't really contain any additional functionality.

**Review by manish99verma — COMMENTED** [review]
Removed Custom timer class. I thought it's easier to manage start and pause  with custom timer class.

File: app/src/main/java/com/github/libretube/ui/fragments/PlayerFragment.kt [hunk 1]
```diff
@@ -252,6 +251,9 @@ class PlayerFragment : Fragment(), OnlinePlayerOptions {
         }
     }
 
+    // schedule task to save the watch position each second
+    private var watchPositionTimer = Timer()
```
Comment (Bnyro) [inline app/src/main/java/com/github/libretube/ui/fragments/PlayerFragment.kt:255]:
```suggestion
    private var watchPositionTimer: Timer? = null
```
We don't need to initialize the timer here - it's going to be overwritten before the first usage anyways.

File: app/src/main/java/com/github/libretube/ui/fragments/PlayerFragment.kt [hunk 2]
```diff
@@ -276,6 +278,18 @@ class PlayerFragment : Fragment(), OnlinePlayerOptions {
                     100
                 )
             }
+
+            //Start or pause watch position timer
```
Comment (Bnyro) [inline app/src/main/java/com/github/libretube/ui/fragments/PlayerFragment.kt:282]:
```suggestion
            // Start or pause watch position timer
```

File: app/src/main/java/com/github/libretube/ui/fragments/PlayerFragment.kt [hunk 3]
```diff
@@ -907,7 +912,7 @@ class PlayerFragment : Fragment(), OnlinePlayerOptions {
 
             val videoStream = streams.videoStreams.firstOrNull()
             val isShort = PlayingQueue.getCurrent()?.isShort == true ||
-                (videoStream?.height ?: 0) > (videoStream?.width ?: 0)
+                    (videoStream?.height ?: 0) > (videoStream?.width ?: 0)
```
Comment (Bnyro) [inline app/src/main/java/com/github/libretube/ui/fragments/PlayerFragment.kt:915]: Please undo that formatting change.

File: app/src/main/java/com/github/libretube/ui/fragments/PlayerFragment.kt [hunk 4]
```diff
@@ -945,7 +950,7 @@ class PlayerFragment : Fragment(), OnlinePlayerOptions {
                 if (binding.playerMotionLayout.progress != 1.0f) {
                     // show controllers when not in picture in picture mode
                     val inPipMode = PlayerHelper.pipEnabled &&
-                        PictureInPictureCompat.isInPictureInPictureMode(requireActivity())
+                            PictureInPictureCompat.isInPictureInPictureMode(requireActivity())
```
Comment (Bnyro) [inline app/src/main/java/com/github/libretube/ui/fragments/PlayerFragment.kt:953]: Please undo that formatting change.

**Comment by Bnyro** [comment]
Just some nitpick comments, the idea generally looks good to me :+1:

**Review by Bnyro — APPROVED** [review]
Thank you!

**Review by manish99verma — COMMENTED** [review]
Sorry for making those silly mistakes. 
Your suggestions really helps me to improve my code style.

**Pull request merged**